dev_requirements.txt

```diff
@@ -11,13 +11,8 @@ flake8~=3.7.9
 black~=19.10b0
 
 # Documentation and examples.
-sphinx~=3.5.3
-docutils==0.16
-sphinxcontrib-bibtex~=1.0.0
-sphinx-copybutton~=0.2.11
-sphinx-autodoc-typehints~=1.10.3
-myst-nb~=0.11.1
-myst-parser~=0.13.3
-pybtex~=0.22.2
-pydata-sphinx-theme~=0.3.1
-notebook~=6.1.5
+sphinxcontrib-bibtex
+sphinx-copybutton
+sphinx-autodoc-typehints
+myst-nb
+pydata-sphinx-theme
```

Comment:
As decided in a previous discussion, if possible, we should pin the versions of the requirements. Maybe you could put the versions that you have in your local environment.

Reply:
I think the present situation is a clear argument not to do this, but ok.

Reply:
Requirements in general should be pinned, agree with @andreamari. It looks like we shouldn't have been using this feature in the first place as it was deprecated in v2.4 of sphinx: https://github.com/agronholm/sphinx-autodoc-typehints/pull/138#issue-412192259. I guess I don't see unpinning things would fix issues like this, though I may be missing something.